### PR TITLE
1103 release matrix core key

### PR DESCRIPTION
**Pull request by daveatweaviate**
[issue](https://github.com/weaviate/weaviate-io/issues/1103)
[staging site](https://6504ba9608aa41259b900b9f--tangerine-buttercream-20c32f.netlify.app/developers/weaviate/client-libraries/python#client-releases)

### What's being changed:

Creates a matrix for which client was current when a particular core was released. 

( This PR was moved. The [original PR](https://github.com/weaviate/weaviate-io/pull/1106) is now closed. )

### Type of change:

- [x] **Documentation** updates (non-breaking change to fix/update documentation)

### How Has This Been Tested?

- [x] **Github action** – automated build completed without errors  (Unrelated link failure, looks like a timeout)
- [x] **Local build** - the site works as expected when running `yarn start`

### CODE REVIEW

File: _includes/more-resources-docs.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
-If you can't find the answer to your question here, please look at the:
+For additional information, try these sources.
```
Comment (daveatweaviate) [inline _includes/more-resources-docs.md:1]: Reformatted to get rid of the Ors at the end of the lines. 

Should we consider dropping this section, or limiting it to Weaviate sources?

File: developers/weaviate/client-libraries/go.md [hunk 2]
```diff
@@ -397,12 +397,19 @@ After:
...
 
 ## Change logs
 
-Check the [change logs on GitHub](https://github.com/weaviate/weaviate-go-client/releases) for updates on the latest `Go client` changes.
+Check the
+[change logs on GitHub](https://github.com/weaviate/weaviate-go-client/releases)
+for updates on the latest `Go client` changes.
 
-## More Resources
+## More resources
```
Comment (daveatweaviate) [inline developers/weaviate/client-libraries/go.md:412]: Made this sentence case to match the rest of the page

**Review by daveatweaviate — COMMENTED** [review]
Notes for the reviewer

**Comment by daveatweaviate** [comment]
@databyjp I closed the original PR so I could chagne repos and get staging builds.  I also swapped the chart so it keys off the core releases. Would you have a look?

**Comment by databyjp** [comment]
Thanks @daveatweaviate, looks great.

Would you be able to add rows for `1.21` at the top? Then I'm happy to merge.

File: _includes/clients/matrix-intro.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+This chart matches Weaviate database releases with Weaviate client releases. It
+lists the most recent database version when each client version was released.
+
+The chart includes point releases for the most recent major and minor versions
+of the client. Earlier client releases have less detailed version information.
+
+
+|Weaviate Version|Release Date|Python|TypeScript|Go|Java|
+|:-|:-|:-|:-|:-|:-|
```
Comment (databyjp) [inline _includes/clients/matrix-intro.md:9]: The 1.21 notes should go here imo

Comment (daveatweaviate) [reply _includes/clients/matrix-intro.md:9]: Updated.

**Review by daveatweaviate — COMMENTED** [review]
@databyjp Thanks for the feedback, this is ready for another look

[updated staging](https://6508608a91bdff04eacc52ba--tangerine-buttercream-20c32f.netlify.app/developers/weaviate/client-libraries/python#client-releases)

**Comment by databyjp** [comment]
Merged! :) 

For future reference - it might be worth considering keeping only the client version names where they've changed. 

So this:

|[1.21.3](https://github.com/weaviate/weaviate/releases/tag/v1.21.3)|2023-09-13|3.24.1|1.5.0|4.10.0|4.3.0|
|[1.21.2](https://github.com/weaviate/weaviate/releases/tag/v1.21.2)|2023-08-30|3.23.1|1.5.0|4.10.0|4.3.0|
|[1.21.1](https://github.com/weaviate/weaviate/releases/tag/v1.21.1)|2023-08-22|3.23.0|1.5.0|4.10.0|4.3.0|
|[1.21.0](https://github.com/weaviate/weaviate/releases/tag/v1.21.0)|2023-08-17|3.22.1|1.4.0|4.9.0|4.2.1|
|[1.20.0](https://github.com/weaviate/weaviate/releases/tag/v1.20.0)|2023-07-06|3.22.0|1.4.0|4.9.0|4.2.0|

would become:

|[1.21.3](https://github.com/weaviate/weaviate/releases/tag/v1.21.3)|2023-09-13|3.24.1| - | - | - |
|[1.21.2](https://github.com/weaviate/weaviate/releases/tag/v1.21.2)|2023-08-30|3.23.1| - | - | - |
|[1.21.1](https://github.com/weaviate/weaviate/releases/tag/v1.21.1)|2023-08-22|3.23.0|1.5.0|4.10.0|4.3.0|
|[1.21.0](https://github.com/weaviate/weaviate/releases/tag/v1.21.0)|2023-08-17|3.22.1| - | - |4.2.1|
|[1.20.0](https://github.com/weaviate/weaviate/releases/tag/v1.20.0)|2023-07-06|3.22.0|1.4.0|4.9.0|4.2.0|

Wdyt?

**Pull request merged**